internal/webhooks/cluster.go

```diff
@@ -170,6 +170,28 @@ func (webhook *Cluster) validate(ctx context.Context, oldCluster, newCluster *cl
 			),
 		)
 	}
+	if newCluster.Spec.ClusterNetwork != nil {
+		if newCluster.Spec.ClusterNetwork.Pods != nil && len(newCluster.Spec.ClusterNetwork.Pods.CIDRBlocks) > 2 {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					specPath.Child("clusterNetwork", "pods", "cidrBlocks"),
```

Comment:
+1, I think it would make sense to check that either:
1) 1 ipv4 block is specified
2) 1 ipv6 block is specified
3) 1 ipv4 and 1 ipv6 blocks are specified

are we already checking that the CIDR blocks are valid IP ranges?

Reply:
Not at the webhook - there's currently no validation for these fields. At this point it's treated as just an array of strings and Cluster creation will continue until something goes wrong. With a managed cluster this is eventually checked during reconciliation and it causes an error.